external-crates/move/crates/move-compiler/src/command_line/compiler.rs

```diff
@@ -54,6 +54,8 @@ pub struct Compiler<'a> {
     known_warning_filters: Vec<(/* Prefix */ Option<Symbol>, Vec<WarningFilter>)>,
     package_configs: BTreeMap<Symbol, PackageConfig>,
     default_config: Option<PackageConfig>,
+    /// Abstracted source file reader
+    source_file_reader: Box<dyn SourceFileReader>,
```

Comment:
Nit, make this a `Option`, and assert it is None when being set so that it is not set more than once

Reply:
I did it in a similar style to how other `Options` in the compiler are handled, but I am not sure if it's the most elegant (though at least consistent). Another alternative perhaps would be to make `Option<Box<dyn SourceFileReader>>` an argument to `from_package_paths`.

Reply:
I think just making this an Option here would be fine, and would match the other ones like `interface_files_dir_opt`, `pre_compiled_lib`, or `package_configs`

Reply:
This is indeed the solution I went for.